### PR TITLE
feat(core): input plugin

### PR DESCRIPTION
**Pull request by ripel2**
Related to:
- #15 

I've implemented a very simple wrapper for GLFW GetKey function.
I also had to implement window handling in this issue to allow the usage of GLFW functions

No unit tests because it is (almost) impossible to test this kind of stuff

(see below for test main)

### CODE REVIEW

**Comment by ripel2** [comment]
Test main function:

```cpp
#include <GLFW/glfw3.h>
#include <iostream>
#include "Registry.hpp"
#include "window/resource/Window.hpp"
#include "input/resource/InputManager.hpp"

int main()
{
    ES::Engine::Registry registry;

    registry.RegisterResource<ES::Plugin::Window::Resource::Window>({800, 600, "GLFW Window"});
    registry.RegisterResource<ES::Plugin::Input::Resource::InputManager>({});

    ES::Plugin::Window::Resource::Window &window = registry.GetResource<ES::Plugin::Window::Resource::Window>();
    ES::Plugin::Input::Resource::InputManager &inputManager = registry.GetResource<ES::Plugin::Input::Resource::InputManager>();

    window.Create();

    while (window.ShouldClose() == false)
    {
        glfwPollEvents();
        glfwSwapBuffers(window.GetGLFWWindow());

        if (inputManager.IsKeyPressed(GLFW_KEY_Z))
        {
            std::cout << "Z key pressed" << std::endl;
        }

        if (inputManager.IsMouseButtonPressed(GLFW_MOUSE_BUTTON_LEFT))
        {
            std::pair<double, double> mousePos = inputManager.GetMousePosition();
            std::cout << "mouse pos: " << mousePos.first << ", " << mousePos.second << std::endl;
        }
    }

    return 0;
}```

File: src/plugin/input/resource/InputManager.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <GLFW/glfw3.h>
+
+#include <functional>
+#include <map>
+#include <utility>
+
+#include "Registry.hpp"
+#include "Window.hpp"
```
Comment (Miou-zora) [inline src/plugin/input/resource/InputManager.hpp:10]:
```suggestion
#include <utility>

#include "Registry.hpp"
```

File: src/plugin/window/resource/Window.hpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    class Window
+    {
+        private:
+            int _width;
+            int _height;
+            std::string _title;
+            GLFWwindow* _window;
+            GLFWmonitor* _monitor;
+            GLFWwindow* _share;
```
Comment (Miou-zora) [inline src/plugin/window/resource/Window.hpp:22]: Do you think it's better to use raw ptr or smart ptr ?

Comment (Miou-zora) [inline src/plugin/window/resource/Window.hpp:22]: Do we put members attributes at the start or at the end of the class definition ?

File: src/plugin/window/resource/Window.hpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                _window = glfwCreateWindow(_width, _height, _title.c_str(), _monitor, _share);
+
+                if (!_window)
+                {
+                    glfwTerminate();
+                    throw ES::Plugin::Window::Exception::WindowError("Failed to create GLFW window");
+                }
+
+                glfwMakeContextCurrent(_window);
+            }
```
Comment (Miou-zora) [inline src/plugin/window/resource/Window.hpp:51]: I think that you should put this inside a cpp file

File: src/plugin/window/resource/Window.hpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+            ~Window()
+            {
+                if (_window)
+                {
+                    glfwDestroyWindow(_window);
+                    glfwTerminate();
+                }
+            }
```
Comment (Miou-zora) [inline src/plugin/window/resource/Window.hpp:80]: Same comment as `void Create()` method

**Review by Miou-zora — CHANGES_REQUESTED** [review]

File: src/plugin/window/resource/Window.hpp (hunk 2 above)
Comment (ripel2) [reply src/plugin/window/resource/Window.hpp:22]: we'll see with the coding style issue #13, right now it doesn't matter
(i personnaly prefer declaring them on top)

File: src/plugin/window/resource/Window.hpp (hunk 3 above)
Comment (ripel2) [reply src/plugin/window/resource/Window.hpp]: done

File: src/plugin/window/resource/Window.hpp (hunk 4 above)
Comment (ripel2) [reply src/plugin/window/resource/Window.hpp]: done

File: src/plugin/window/resource/Window.hpp (hunk 2 above)
Comment (Miou-zora) [reply src/plugin/window/resource/Window.hpp:22]: ok good 👍

**Pull request merged**